src/js/core/MesosStream.js

```diff
@@ -3,5 +3,6 @@ import { ReplaySubject } from "rxjs/ReplaySubject";
 
 export const MesosStreamType = Symbol("MesosStreamType");
 export default stream({ type: "SUBSCRIBE" }, "/mesos/api/v1?subscribe")
+  .repeat()
```

Comment:
We are not talking about [RxJS.Obersvable.repeat](https://github.com/Reactive-Extensions/RxJS/blob/master/src/core/perf/operators/repeat.js) here, right?

Reply:
We're talking exactly about that :) When MesosStream closes the connection the observable completes and we need to repeat it

Reply:
Okay, this makes sense, if no new data can be fetched because some service is down, just show the old state, do I get that right?

Does `stream` re-open / retry the HTTP Request or how does that part work?

Reply:
It will just attempt to reconnect.
[As you can see](https://github.com/dcos-labs/http-service/blob/master/src/stream.js#L20) in the `http-client` if connection completes then observer completes which means it's done, no new data. This is where `repeat` kicks in, so it will eventually open a new connection.

I'd be happy guiding you through the `http-client` code

Reply:
Oh, so repeat does not mean "send the same value as the last value" but "call the same observable again"? I understood the docs differently, but I am also new to RxJS